R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
+#'                                          remove_chr_prefix = FALSE, 
+#'                                          from_flatfile = TRUE, 
+#'                                          streamlined = FALSE) 
+#'                                     
+#' #Example using the regions parameter:
+#' segments_region_hg38 = get_cn_segments(region = "chr8:128,723,128-128,774,067",
+#'                                        projection = "hg38", 
+#'                                        this_seq_type = "genome", 
+#'                                        remove_chr_prefix = TRUE, 
+#'                                        from_flatfile = TRUE, 
+#'                                        streamlined = FALSE) 
+#'
+#'
+get_cn_segments = function(region,
+                           chromosome,
                            qstart,
                            qend,
-                           region,
-                           with_chr_prefix = FALSE,
+                           projection = "grch37",
+                           this_seq_type = "genome",
+                           remove_chr_prefix = FALSE,
                            streamlined = FALSE,
-                           from_flatfile = FALSE){
+                           from_flatfile = TRUE){
+  
+  #checks
   remote_session = check_remote_configuration()
-  db = config::get("database_name")
-  table_name = config::get("results_tables")$copy_number
-  table_name_unmatched = config::get("results_tables")$copy_number_unmatched
+  
+  #check seq type and return a message if anything besides "genome" is called. To be updated once capture samples have been processed through CNV protocols.
+  if(this_seq_type!="genome"){
+    stop("Currently, only genome samples are available for this function. Please select a valid seq type (i.e genome). Compatibility for capture samples will be added soon...")
+  }
+  
+  #get wildcards from this_seq_type (lazy)
+  seq_type = this_seq_type
+  
+  #perform wrangling on the region to have it in the correct format. 
   if(!missing(region)){
     region = gsub(",", "", region)
-    #format is chr6:37060224-37151701
     split_chunks = unlist(strsplit(region, ":"))
     chromosome = split_chunks[1]
     startend = unlist(strsplit(split_chunks[2], "-"))
     qstart = startend[1]
     qend = startend[2]
   }
-  if(grepl("chr", chromosome)){
+  
+  #deal with chr prefixes for region, based on selected genome projection.
+  if(projection == "grch37"){
+    if(grepl("chr", chromosome)){
+      chromosome = gsub("chr", "", chromosome)
+    }
   }else{
-    chromosome = paste0("chr", chromosome)
+    if(!grepl("chr", chromosome)){
+      chromosome = paste0("chr", chromosome)
+    }
   }
-
-  #chr prefix the query chromosome to match how it's stored in the table.
-  #This isn't yet standardized in the db so it's just a workaround "for now".
+  
+  #sanity check type of qstart en qend
+  if(!is.numeric(qstart)){
```

Comment:
The ifelse checks can be resource consuming on large data frames with lots of rows. [Here is a different intention ](https://stackoverflow.com/a/41585689/16737475) but it shows that the checks are very slow. Instead, nothing bad will happen if we just always mutate the qstart and qend to numeric regardless of what they initially are.

Reply:
good point